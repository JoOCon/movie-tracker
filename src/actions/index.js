export const addMovies = (movieData) => ({
  type: 'ADD_MOVIES',
  movieData
});

export const addFavorite = (movie) => ({
  type: 'ADD_FAVORITE',
  movie
});

export const setActiveUser = ({ id, name }) => ({
  type: 'SET_ACTIVE_USER',
  id,
  name
});

export const setUserFavorites = (favorites) => ({
  type: 'SET_USER_FAVORITES',
  favorites
});