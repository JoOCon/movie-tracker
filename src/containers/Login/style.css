.log-header {
  color: #ffcb9a; }

.user-login {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  position: absolute;
  left: 35.5%;
  top: 20%;
  z-index: 1;
  background-color: rgba(44, 53, 49, 0.794);
  height: 20em;
  width: 26em;
  font-family: "Montserrat", sans-serif; }

.input-field {
  height: 2em;
  width: 60%;
  margin-bottom: 1.5em;
  font-size: .85em;
  font-weight: bold; }
