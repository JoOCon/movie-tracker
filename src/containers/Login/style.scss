.user-login {
  align-items: center;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  position: absolute;
  left: 39%;
  top: 20%;
  z-index: 1;
  background-color: pink;
  height: 20em;
  width: 26em;
}

.input-field {
  height: 2em;
  width: 60%;
}