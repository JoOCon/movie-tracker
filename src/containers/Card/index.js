import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import StarRatings from 'react-star-ratings';
import { addFavorite } from '../../actions';
import { postFavorites } from '../../api/apiCalls';
import './style.css';

export const Card = (props) => {
  const { movie, user, addFavorite } = props;
  const { poster_path, vote_average } = movie;
  const rating = vote_average / 2;

  const handleFavorite = () => {
    const favoriteData = {...movie, user_id: user.id};
    addFavorite(favoriteData);
    postFavorites(favoriteData);
  };

  return (
    <div className='card'>
      <div className='favorite-wrapper'>
        <button 
          className='favorite' 
          onClick={() => handleFavorite()}>
        </button>
        <StarRatings 
          starDimension={'1em'} 
          rating={rating} 
          numberOfStars={5} 
          starRatedColor={'gold'} 
        />
      </div>
      <img className='poster-image' src={poster_path} alt='movie data' />
    </div>
  );
};

const mapStateToProps = (state) => ({
  user: state.userData
});

const mapDispatchToProps = (dispatch) => ({
  addFavorite: (movie) => dispatch(addFavorite(movie)),
  postFavorites: (movie) => dispatch(postFavorites(movie))
});

Card.propTypes = {
  movie: PropTypes.object,
  user: PropTypes.object,
  addFavorite: PropTypes.func,
  postFavorites: PropTypes.func
};

export default connect(mapStateToProps, mapDispatchToProps)(Card);
