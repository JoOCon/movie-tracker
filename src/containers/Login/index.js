import React, { Component } from 'react';
import './style.css';
import { login } from '../../api/apiCalls'

export class Login extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = (e) => {
    e.preventDefault();
    const { name, value } = e.target;
    this.setState({ [name]: value });
  }

  handleSubmit = async (e) => {
    e.preventDefault();
    await login(this.state);
    // this.setState({ userName: '', password: '' });
  }

  render() {
    const { email, password } = this.state;
    return (
      <form className='user-login' onSubmit={this.handleSubmit}>
        <input name='email' placeholder='email' type='email' value={email} onChange={this.handleChange} />
        <input name='password' placeholder='password' type='password' value={password} onChange={this.handleChange} />
        <button>LogIn</button>
      </form>
    );
  }
}
