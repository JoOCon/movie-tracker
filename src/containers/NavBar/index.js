import React from 'react';
import { Link } from 'react-router-dom';
import './style.css';

export const NavBar = () => {
  return (
    <div className='nav-bar'>
      <Link to='/login'><button className='login'>login</button></Link>
      <Link to='/signup'><button className='signup'>SignUp</button></Link>
    </div>
  );
};