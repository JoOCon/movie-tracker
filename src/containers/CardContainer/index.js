import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Card from '../../components/Card';
import './style.css';

const CardContainer = (props) => {
  const displayMovies = props.movies.map(movie => <Card key={movie.id} movie={movie} />);
  return (
    <div className="card-container">
      {displayMovies}
    </div>
  );
};

const mapStateToProps = (state) => ({
  movies: state.moviesData
});

CardContainer.propTypes = {
  movies: PropTypes.array
};

export default connect(mapStateToProps)(CardContainer);
