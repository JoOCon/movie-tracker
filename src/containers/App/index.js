import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { BrowserRouter, Route } from 'react-router-dom';
import { addMovies } from '../../actions';
import { fetchMovies } from '../../api/apiCalls';
import CardContainer from '../CardContainer';
import FavoriteContainer from '../FavoriteContainer';
import Login from '../../containers/Login';
import SignUp from '../../containers/SignUp';
import { Header } from '../../components/Header';
import './style.css';

export class App extends Component {
  componentDidMount = () => {
    this.fetchOpeningMovies();
  }

  fetchOpeningMovies = async () => {
    const { addMovies } = this.props;
    const openingMovies = await fetchMovies();
    addMovies(openingMovies);
  }

  render() {
    return (
      <BrowserRouter>
        <div className="App">
          <Route path='/' component={Header} />
          <Route exact path='/login' component={Login} />
          <Route exact path='/signup' component={SignUp} />
          <Route exact path='/' component={CardContainer} />
          <Route exact path='/:landingPage' component={CardContainer} />
          <Route exact path='/:user/favorites' component={FavoriteContainer} />
        </div>
      </BrowserRouter>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  addMovies: (moviesData) => dispatch(addMovies(moviesData))
});

App.propTypes = {
  addMovies: PropTypes.func
};

export default connect(null, mapDispatchToProps)(App);
