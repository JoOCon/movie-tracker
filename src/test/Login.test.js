import React from 'react';
import { Login, mapStateToProps, mapDispatchToProps } from '../containers/Login';
import { shallow } from 'enzyme';
import * as mockData from '../test/mockData';
import * as actions from '../actions';

describe('Login tests', () => {
  let wrapper;
  let mockSetActiveUser;
  let mockSetUserFavorites;
  const { mockUserData } = mockData;
  beforeEach(() => {
    mockSetActiveUser = jest.fn();
    mockSetUserFavorites = jest.fn();
    wrapper = shallow(
      <Login
        user={mockUserData}
        setActiveUser={mockSetActiveUser}
        setUserFavorites={mockSetUserFavorites}
      />
    );
  });

  it('renders the Card with the correct props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('has a default state of email, password and fireRedirect property', () => {
    expect(wrapper.state('email')).toEqual('');
    expect(wrapper.state('password')).toEqual('');
    expect(wrapper.state('fireRedirect')).toEqual(false);
  });

  it('should change state when email input is changed', () => {
    const mockEvent = { target: { value: 'mock@mock.com', name: 'email' }, preventDefault: jest.fn() };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('email')).toEqual(mockEvent.target.value);
  });

  it('should change state when password input is changed', () => {
    const mockEvent = { target: { value: 'mockpassword', name: 'password' }, preventDefault: jest.fn() };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('password')).toEqual(mockEvent.target.value);
  });

  describe('handleSubmit', () => {
    let mockEvent;
    beforeEach(() => {
      mockEvent = { preventDefault: jest.fn(), target: { email: '', password: '' } };
    });

    it.skip('should setActive user when recieving a valid email and password', async () => {
      await wrapper.find('.user-login').simulate('submit', mockEvent);
      expect().toHaveBeenCalled();
    });
  });

  describe('MapStateToProps', () => {
    const { mockUserData } = mockData;
    it('should have a users data object in props', () => {
      const mockState = {userData: mockUserData};
      const expected = {user: mockUserData};
      const props = mapStateToProps(mockState);
      expect(props).toEqual(expected);
    });
  });

  describe('MapDispatchToProps', () => {
    let mockDispatch;
    beforeEach(() => {
      mockDispatch = jest.fn();
    });

    it('should dispatch setActiveUser when called', () => {
      const { mockUserData } = mockData;
      const { setActiveUser } = actions;
      const mockAction = setActiveUser(mockUserData);
      const props = mapDispatchToProps(mockDispatch);
      props.setActiveUser(mockUserData);
      expect(mockDispatch).toBeCalledWith(mockAction);
    });

    it('should dispatch setUserFavorites when called', () => {
      const { mockCleanMoviesDataArray } = mockData;
      const { setUserFavorites } = actions;
      const mockAction = setUserFavorites(mockCleanMoviesDataArray);
      const props = mapDispatchToProps(mockDispatch);
      props.setUserFavorites(mockCleanMoviesDataArray);
      expect(mockDispatch).toBeCalledWith(mockAction);
    });
  });
});
