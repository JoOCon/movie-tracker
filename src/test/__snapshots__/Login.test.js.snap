// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login tests renders the Card with the correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Login
    mockSetUserFavorites={[Function]}
    setActiveUser={[Function]}
    user={
        Object {
            "name": "Taylor",
            "user_id": 1,
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="user-login"
          onSubmit={[Function]}
>
          <h2
                    className="log-header"
          >
                    Log In
          </h2>
          <input
                    className="email input-field"
                    name="email"
                    onChange={[Function]}
                    placeholder="email"
                    type="email"
                    value=""
          />
          <input
                    className="password input-field"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
          />
          <button
                    className="log-button"
          >
                    Log In
          </button>
</form>,
        <Link
          replace={false}
          to="/signup"
>
          <button
                    className="log-button"
          >
                    Sign UP
          </button>
</Link>,
        <Link
          replace={false}
          to="/"
>
          <button
                    className="log-button send-home"
          >
                    Home
          </button>
</Link>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2
              className="log-header"
>
              Log In
</h2>,
            <input
              className="email input-field"
              name="email"
              onChange={[Function]}
              placeholder="email"
              type="email"
              value=""
/>,
            <input
              className="password input-field"
              name="password"
              onChange={[Function]}
              placeholder="password"
              type="password"
              value=""
/>,
            <button
              className="log-button"
>
              Log In
</button>,
          ],
          "className": "user-login",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "className": "log-header",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "email input-field",
              "name": "email",
              "onChange": [Function],
              "placeholder": "email",
              "type": "email",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "password input-field",
              "name": "password",
              "onChange": [Function],
              "placeholder": "password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Log In",
              "className": "log-button",
            },
            "ref": null,
            "rendered": "Log In",
            "type": "button",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="log-button"
>
            Sign UP
</button>,
          "replace": false,
          "to": "/signup",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign UP",
            "className": "log-button",
          },
          "ref": null,
          "rendered": "Sign UP",
          "type": "button",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <button
            className="log-button send-home"
>
            Home
</button>,
          "replace": false,
          "to": "/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Home",
            "className": "log-button send-home",
          },
          "ref": null,
          "rendered": "Home",
          "type": "button",
        },
        "type": [Function],
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="user-login"
            onSubmit={[Function]}
>
            <h2
                        className="log-header"
            >
                        Log In
            </h2>
            <input
                        className="email input-field"
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        type="email"
                        value=""
            />
            <input
                        className="password input-field"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        type="password"
                        value=""
            />
            <button
                        className="log-button"
            >
                        Log In
            </button>
</form>,
          <Link
            replace={false}
            to="/signup"
>
            <button
                        className="log-button"
            >
                        Sign UP
            </button>
</Link>,
          <Link
            replace={false}
            to="/"
>
            <button
                        className="log-button send-home"
            >
                        Home
            </button>
</Link>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2
                className="log-header"
>
                Log In
</h2>,
              <input
                className="email input-field"
                name="email"
                onChange={[Function]}
                placeholder="email"
                type="email"
                value=""
/>,
              <input
                className="password input-field"
                name="password"
                onChange={[Function]}
                placeholder="password"
                type="password"
                value=""
/>,
              <button
                className="log-button"
>
                Log In
</button>,
            ],
            "className": "user-login",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "className": "log-header",
              },
              "ref": null,
              "rendered": "Log In",
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "email input-field",
                "name": "email",
                "onChange": [Function],
                "placeholder": "email",
                "type": "email",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "password input-field",
                "name": "password",
                "onChange": [Function],
                "placeholder": "password",
                "type": "password",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log In",
                "className": "log-button",
              },
              "ref": null,
              "rendered": "Log In",
              "type": "button",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="log-button"
>
              Sign UP
</button>,
            "replace": false,
            "to": "/signup",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign UP",
              "className": "log-button",
            },
            "ref": null,
            "rendered": "Sign UP",
            "type": "button",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <button
              className="log-button send-home"
>
              Home
</button>,
            "replace": false,
            "to": "/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Home",
              "className": "log-button send-home",
            },
            "ref": null,
            "rendered": "Home",
            "type": "button",
          },
          "type": [Function],
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "disableLifecycleMethods": true,
    "hydrateIn": undefined,
  },
}
`;
